### DISCUSSION TITLE
How do I run SLAM on data generated in AirSim?

### DISCUSSION THREAD

**Discussion by RitvikN** [microsoft/AirSim#4450] [Support Q&A]
Hi, I would appreciate some guidance on how to run SLAM on training data generated by flying a drone in one of AirSim's environments. I'm a complete beginner and really don't know where to start.

I haven't been any to find any tutorials online on how to accomplish this. Looking at some off the issues raised on this GitHub and the comments posted, it looks like several people have been able to get Orb SLAM working with AirSim, but no one has posted the steps on how to actually do it.

I took a look at the Orb SLAM repo and it has several scripts which they have created to run with various datasets like KITTI. Can the same thing be done with AirSim, if I fly a drone around in an env and record the data?

How about if I want to run SLAM in real time while the drone is flying in the AirSim environment? It looks like I need to have the odometry data published on ROS and have a script using that data for SLAM. Unfortunately I cannot find any tutorials to follow online.

It would be extremely helpful if anyone could point me in the right direction. Thank you.

**Comment by ivannson** [comment]
You can do it, just treat AirSim as any other real camera/sensor set up. The only tricky thing for me was getting the camera intrinsic matrix, but I think it is now obtainable directly from airsim. I think it would be easier to use ROS if you want to do it in real time. 

There are plenty of tutorials available, just don't look for tutorials on how to do SLAM specifically in AirSim - you won't find any. 

For example, I think I was able to replicate [this article](https://medium.com/@j.zijlmans/orb-slam-2052515bd84c) in AirSim.

**Comment by Nikhil-Sudhan** [comment]
I am looking for the same thing. If  you find any articles or related to the topic please share here. Or you can tell how to share the data from unreal engine and what are the software your using. Thank you!
